view/frontend/web/js/view/payment/method-renderer/razorpay-method.js

```diff
@@ -153,6 +161,85 @@ define(
                 });
             },
 
+            _prepareHiddenFormFields: function (options, form) {
+                var self = this;
+
+                function visitNestedOption(options, parentKey) {
+                    for (let curKey in options) {
+                      if (options.hasOwnProperty(curKey)) {
+                        const value = options[curKey];
+                        let prepareKey = parentKey ? `${parentKey}[${curKey}]` : curKey;
+
+                        if (typeof value === 'object') {
+                          visitNestedOption(value, prepareKey);
+                        } else {
+                          // Exception: Rename key -> key_id (merchant key)
+                          if (prepareKey === 'key') {
+                            prepareKey = 'key_id';
+                          }
+
+                          form.appendChild(self._createHiddenInput(prepareKey, value));
+                        }
+                      }
+                    }
+                }
+              visitNestedOption(options);
+            },
+
+            _createHiddenInput: function(key, value) {
+              var input = document.createElement('input');
+
+              input.type = 'hidden';
+              input.name = key;
+              input.value = value;
+
+              return input;
+            },
+
+            renderHosted: function(data) {
+                var self = this;
+
+                this.merchant_order_id = data.order_id;
+
+                var opts = {
+                    key: self.getKeyId(),
+                    name: self.getMerchantName(),
+                    amount: data.amount,
+                    order_id: data.rzp_order,
+                    notes: {
+                        merchant_order_id: '',
+                        merchant_quote_id: data.order_id
+                    },
+                    prefill: {
+                        name: this.user.name,
+                        contact: this.user.contact,
+                        email: this.user.email
+                    },
+                    callback_url: url.build('razorpay/payment/order'),
+                    cancel_url  : url.build('checkout/cart'),
+                    _: {
+                        integration: 'magento',
+                        integration_version: data.module_version,
+                        integration_parent_version: data.maze_version,
+                    }
+                }
+                const options = JSON.parse(JSON.stringify(opts));
+
+                var form = document.createElement('form'),
+                    method = 'POST',
+                    input,
+                    key;
+
+                form.method = method;
+                form.action = 'https://api.razorpay.com/v1/checkout/embedded';
+
+                self._prepareHiddenFormFields(options, form);
```

Comment:
what is we directly call "visitNestedOption" from here.

Reply:
As we need to process object fields also here (prefill, notes) I had done like this.